source/main/providers/citeproc-provider.js

```diff
@@ -278,6 +278,7 @@ class CiteprocProvider {
       // is only for preview purposes, it should follow the language like the
       // rest of the interface.
       this._engine = new citeproc.Engine(this._sys, this._mainStyle, this._lang)
+      this._engine.opt.development_extensions.wrap_url_and_doi = true
```

Comment:
As it looks like a `development_extension`, is there documentation on the stability of this approach?

Reply:
As far as I can tell there is no official documentation here, this whole thing is incredibly sketchy to me. But it works!

Here's the code in citeproc-js that implements it:
https://github.com/Juris-M/citeproc-js/blob/3389e9b23a1873c1bb0af16eef7533aefd669701/src/node_text.js#L320-L365

Reply:
Hmhmhmhm … I'm a little bit unsure. I REALLY see the wish (and all the benefits) of making URLs clickable, but I'm afraid we're breaking something using this (more or less) undocumented code …

So I basically see two options:

1. Implement something that simply auto-links any link in the rendered bibliography (hence, in the `zettlr-attachments.js`-file), which would be fine, because then it's "our" code (and we don't depend on another library)
2. Implement it this way, but then we need a switch, because I don't want to risk relying on features from a third-party-library that are clearly labeled unstable (because this way, in case something breaks we can simply tell people that they should uncheck that option in the preference until a fix)

So one of the two approaches would work, but I'll leave the one you choose open to you because it's your PR! :)

I personally am biased towards option 1 (which would be much more stable and easier to implement) – here's how we might achieve that:

https://github.com/Zettlr/Zettlr/blob/e64a4a7b033f6db552e0cada428c4bd275ddadd3/source/main/commands/update-check.js#L110-L116

Just instead of transforming already HTML-links take something that checks URLs!

Reply:
Makes sense! I don't know if I'm going to have the time to implement this, but I donate my PR to y'all if you're interested in taking up the banner :)

Reply:
Take your time, it's your hard work, and this means you're in control :) It'll take some time either way until the next version will be released!